pkg/ccl/importccl/exportcsv_test.go

```diff
@@ -319,108 +337,168 @@ func validateParquetFile(t *testing.T, file string, test parquetTest) error {
-		// insufficient: many go native types are encoded as []byte, so in the
-		// future, each column will have to call it's own custom decoder ( this is
-		// how IMPORT Parquet will work).
-		decodedV = string(vv)
-	case int64, float64, bool:
-		decodedV = vv
-	case map[string]interface{}:
-		var arrDecodedV []interface{}
-		b := vv["list"].([]map[string]interface{})
-
-		// Array values are stored in an array of maps: []map[string]interface{},
-		// where the ith map contains a single key value pair. The key is always "element"
-		// and the value is the ith value in the array.
-
-		// If the array of maps only contains an empty map, the array is empty. This
-		// occurs IFF "element" is not in the map.
-
-		// NB: there's a bug in the fraugster-parquet vendor library around reading
-		// an ARRAY[NULL]. See the "arrays" test case for more info. Ideally, once
-		// the bug gets fixed, ARRAY[NULL] will get read as the kvp {"element":interface{}} while
-		// ARRAY[] will continue to get read as an empty map.
-		if _, nonEmpty := b[0]["element"]; !nonEmpty {
-			arrDecodedV = []interface{}{}
-			if len(b) > 1 {
-				t.Fatalf("array is empty, it shouldn't have a length greater than 1")
-			}
-		} else {
-			// For non-empty arrays
-			for _, elMap := range b {
-				arrDecodedV = append(arrDecodedV, decodeEl(t, elMap["element"]))
+func TestRandomParquetExports(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	dir, dirCleanupFn := testutils.TempDir(t)
+	defer dirCleanupFn()
+	dbName := "rand"
+
+	params := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			UseDatabase:   dbName,
+			ExternalIODir: dir,
+		},
+	}
+	ctx := context.Background()
+	tc := testcluster.StartTestCluster(t, 3, params)
+	defer tc.Stopper().Stop(ctx)
+	sqlDB := sqlutils.MakeSQLRunner(tc.Conns[0])
+
+	sqlDB.Exec(t, fmt.Sprintf("CREATE DATABASE %s", dbName))
+
+	tableName := "table_1"
+	// TODO (butler): Randomly generate additional table(s) using tools from PR
+	// #75677. The function below only creates a deterministic table with a bunch
+	// of interesting corner case values.
+	err := randgen.GenerateRandInterestingTable(tc.Conns[0], dbName, tableName)
+	require.NoError(t, err)
+
+	s0 := tc.Server(0)
+	ie := s0.ExecutorConfig().(sql.ExecutorConfig).InternalExecutor
+	{
+		// Ensure table only contains columns supported by EXPORT Parquet
+		_, cols, err := ie.QueryRowExWithCols(
+			ctx,
+			"",
+			nil,
+			sessiondata.InternalExecutorOverride{
+				User:     security.RootUserName(),
+				Database: dbName},
+			fmt.Sprintf("SELECT * FROM %s LIMIT 1", tableName))
+		require.NoError(t, err)
+
+		for _, col := range cols {
+			_, err := importccl.NewParquetColumn(col.Typ, "", false)
+			if err != nil {
+				t.Logf("Column type %s not supported in parquet, dropping", col.Typ.String())
+				sqlDB.Exec(t, fmt.Sprintf(`ALTER TABLE %s DROP COLUMN %s`, tableName, col.Name))
```

Comment:
If you wanted, you could probably build this up into a single schema change. Might be a little faster.

Reply:
huh, so i took a look at the docs for ALTER TABLE ... DROP COLUMN , and it doesn't seem like i can drop multiple columns at once. verified this in demo. Perhaps there's another way to do it, but i think i'll keep it as is.